src/cart/src/gurt/misc.c

```diff
@@ -566,3 +567,88 @@ void d_getenv_int(const char *env, unsigned *int_val)
 	D_DEBUG(DB_TRACE, "d_getenv_int(), get ENV %s as %d.\n", env, value);
 	*int_val = value;
 }
+
+/**
+ * Write formatted data to d_string_buffer_t
+ *
+ * Create a string with the same text that would be created by printf(). The
+ * result string is stored as a C string in a buffer pointed by buf. The
+ * d_string_buffer_t internal buffer grows as text is written.
+ *
+ * \param buf	[IN/OUT] string object where the formatted string will
```

Comment:
doxygen should be \param[in,out], I think.   We need to fix this at some point throughout